pkg/route/allocator/bitmap.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```

Comment:
FYI, this was copied from upstream. It has not changed for months/years. It is a relatively small risk to keep copied vs. vendoring entire Kubernetes to get these. Eventually, we can try to extract these out upstream, but it will require changing the internal API to external API.